lib/auth/Vault.ts

```diff
@@ -356,6 +357,35 @@ export default class Vault {
             });
     }
 
+    /**
+     * A getter for account canonical IDs given a list of account IDs
+     * @param accountIds - list of account IDs
+     * @param log - log object
+     * @param callback - callback function
+     * @returns callback with either error or an object from Vault
+     * containing canonicalID and display name for each account ID
+     */
+    getCanonicalIdsByAccountIds(
+        accountIds: string[],
+        log: RequestLogger,
+        callback: (err: Error | null, data?: AccountCannonicalInfo[]) => void
```

Comment:
nit: this callback is not precisely typed, it cannot have both null error and no data...

type of the callback is more something like `ArsenallBack` (in MongoClientInterface):

```
interface AccountCannonicalInfoCallback {
    (err: null, data: AccountCannonicalInfo[]): void;
    (err: Error): void;
};
```

In any case we should define this type, instead of copying all over the place (or maybe we can use a "standard" generic node error callback...)